task-43

Fargo02 opened the pull request:
- Добавил еще один state (ErrorEmpry)
- Обработал нажатие на кнопку стереть
- Исправил некоторые размеры элементов

Fargo02 committed:
task-43
- Добавил еще один state (ErrorEmpry)
- Обработал нажатие на кнопку стереть
- Исправил некоторые размеры элементов

Fargo02 committed:
task-43
- Убрал state (ErrorEmpry)

sashkir7 commented on app/src/main/java/ru/practicum/android/diploma/ui/search/SearchViewModel.kt:
```diff
@@ -26,6 +26,11 @@ class SearchViewModel(
         useLastParam = true
     ) { changedText -> search(changedText) }
 
+    fun changeStateToEmpty() {
```
Лучше переименовать на clearSearch()

sashkir7 replied:
Не думаю, что в наименовании методов viewModel(а) должны упоминаться слова про state...

sashkir7 commented on app/src/main/java/ru/practicum/android/diploma/ui/search/SearchViewModel.kt:
```diff
@@ -26,6 +26,11 @@ class SearchViewModel(
     ) { changedText -> search(changedText) }
 
+    fun changeStateToEmpty() {
+        val inputState = Input.Empty
+        _state.postValue(VacancyState(inputState, VacanciesList.Empty))
```
Можно сразу заебенить 
_state.postValue(VacancyState(Input.Empty, VacanciesList.Empty))

sashkir7 replied:
То есть можно inputState в переменную не выносить

sashkir7 commented on app/src/main/java/ru/practicum/android/diploma/ui/search/SearchViewModel.kt:
```diff
@@ -43,15 +48,19 @@ class SearchViewModel(
 
-    fun searchDebounce(expression: String) = searchDebounceAction(expression)
+    fun searchDebounce(expression: String) {
+        if (expression != "") {
```
Есть методы isNotBlank
А вообще, можно сделать так

if (expression.isBlank) return
searchDebounceAction(expression)

sashkir7 reviewed (approved):
Оставил ряд комментов

Fargo02 committed:
task-43
- изменил название метода
- добавил expression.isBlank() для проверки непустого запроса

pyclan88 commented on app/src/main/java/ru/practicum/android/diploma/ui/search/SearchFragment.kt:
```diff
@@ -122,11 +117,25 @@ class SearchFragment : BindingFragment<FragmentSearchBinding>() {
+                val inputMethodManager =
+                    requireContext().getSystemService(INPUT_METHOD_SERVICE) as? InputMethodManager
+                inputMethodManager?.hideSoftInputFromWindow(binding.ivEditTextButton.windowToken, 0)
+                binding.etSearch.setText(EMPTY_STRING)
```
я бы так сделал:
binding.etSearch.text.clear()
у моего варианта есть недостатки?

Fargo02 committed:
task-43
- заменил отчистку текста в EditeText

Fargo02 committed:
task-43
- незначительные изменения
